packages/api/endpoints/rules.js

```diff
@@ -167,6 +168,9 @@ async function put(req, res) {
       translatedRule = await translatePostgresRuleToApiRule(pgRule, knex);
       await indexRule(esClient, translatedRule, process.env.ES_INDEX);
     });
+
+    await deleteOldEventSourceMappings(knex, oldApiRule);
```

Comment:
This is another discrepancy between main and feature/rds-phase-3. This logic was removed from the endpoint in the feature branch but it still worked because of the changes below. When the endpoint was switched from using the Rule model to using the rulesHelpers, the `updateRuleTrigger` call contained this same logic.

The catch is that the `updateRuleTrigger` logic in the helper doesn't match what the model was doing in main and had to be updated. This required the endpoint logic to change (revert) as well. 

This is all fairly confusing so happy to talk it through.

Reply:
It also could have been made to work by leaving `deleteOldEventSourceMappings` out of the endpoint and keeping it in the helper. That would have required a lot of test changes though and that made me nervous. Also it would make the merge back to main even more difficult.

Reply:
A couple of questions re: this change -- 

1) Looking at `master`, it looks like there's trigger reversions on error 

```
  } catch (innerError) {
      if (newRule) {
        const ruleWithRevertedTrigger = await ruleModel.updateRuleTrigger(apiRule, oldApiRule);
        await ruleModel.update(ruleWithRevertedTrigger);
      }
      throw innerError;
    }
``

It's not clear to me (yet, still looking into this) why this *wouldn't* be needed in the P3 implementation.      This may or may not be germane to this PR.

2) Looking at the master vs p3 implementation, it looks like the bug might be in L586/599 where it's deleting triggers from the *merged* rule instead of the original, given what `deleteOldEventSourceMappings` was doing in `master`.    I might be missing why we'd want to delete the merged rule *as well* however. 

I'd love to walk through that a bit as I may be mis-reading what's going on here.

Reply:
This might be a bug on main -- does this deletion *remove* ESMs if a user PUTs a rule that's the exact same trigger/etc?

Reply:
^ Confirmed on feature/main, issued CUMULUS-3205 to mitigate/resolve.